src/Components/Generator/DynamicInputContainer.js

```diff
@@ -216,13 +216,34 @@ export default class DynamicInputContainer extends preact.Component {
     }
 
     removeDynamicInput(event) {
-        if (window.confirm(t('confirm-input-remove', 'generator'))) {
-            this.setState(prev => {
-                delete prev.fields[event.target.getAttribute('rel')];
-                return prev;
-            });
-            this.pushStateUp();
+        // prompt only if respective field has value entered while removing
+        if (!this.checkInputData(event)) {
+            this.confirmStateChange(event);
+        } else if(window.confirm(t('confirm-input-remove', 'generator'))) {
+            this.confirmStateChange(event);
+        }   
```

Comment:
I actually don't like to clutter this already cluttered class with another method of very limited use. Try grouping those together in one statement doing something like `!this.checkInputData(…) || window.confirm(…)`.

Reply:
@zner0L Sure, I think what you have suggested totally makes sense. I will update this.